core/src/com/unciv/models/translations/TranslationFileWriter.kt

```diff
@@ -82,8 +87,7 @@ object TranslationFileWriter {
 
             countOfTranslatedLines[language] = translationsOfThisLanguage
 
-            val fileWriter = if (modFolder != null) modFolder.child(languageFileLocation.format(language))
-                                else Gdx.files.local(languageFileLocation.format(language))
+            val fileWriter = getFileHandle(modFolder,languageFileLocation.format(language))
```

Comment:
After looking over this, it looks like I made a mistake.
It's as follows - on Android, there are 2 file locations - internal and local, where internal is the assets packed into the apk and local is files added afterwards, not from the apk.
Internal is read-only, and local is read-write.
In Desktop, where you don't have that separation, they are both in the same location.

HOWEVER, even if you're in Desktop, *by definition* the internal location is read-only, so if you want to edit the assets you need to use the _local_ file handler.

tl;dr local required for writing on Desktop, internal required for reading on Android, which means that this specific part (which does the writing) needs to access the files.local....

My bad

Reply:
I'm curious, though - does it work when you try running it? If it does, then maybe the restriction for "internal is read only" is only active on Android, and the writing code (which is active only on Desktop) can stay as-is

Reply:
I have noticed that sometimes it is `internal` and sometimes it is `local` but did not pay much attention to it since any of these worked well on Desktop.
As far as I understand you, it is enough to replace read-only `internal` for read-write `local`.

Reply:
That should be enough, since we're in the TranslationWriter class
In the TranslationReader we'll need to use `internal`